### PR TITLE
Feat/Select machine

### PR DESCRIPTION
**Pull request by Oolay**
- Adds multi select feature
- Moves all select logic into Select machine that is invoked as a service by the game machine

**Known issue (existing): When two different blobs with selection display selected and one is deselected, the selection display is removed rather than showing the one for the blob still selected.**

### CODE REVIEW

File: src/Game/Game.tsx [hunk 1]
```diff
@@ -66,34 +66,10 @@ export const Game = () => {
...
-      });
-    }
-  };
-
   return (
     <>
       <canvas
         id="game-canvas"
-        onClick={handleMainGameClick}
-        onMouseMove={handleMainGameMouseMove}
```
Comment (Oolay) [inline src/Game/Game.tsx:96]: This is all handled in the Select machine now

File: src/Game/gameMachine/actions/game.ts [hunk 2]
```diff
@@ -28,3 +28,18 @@ export function propagateMapClicked(
...
+  bloblets.forEach((blob) => {
+    blob.send(event);
+  });
+  blobLarvae.forEach((larva) => {
+    larva.send(event);
+  });
+  blobalongs.forEach((blob) => {
+    blob.send(event);
+  });
+}
```
Comment (Oolay) [inline src/Game/gameMachine/actions/game.ts:45]: I think we should propergate the normal Click event in this same way. This will solve the issue we still have where blobs don't deselect when selecting any other blob. Can do this in another PR though

File: src/Game/SelectionDisplay/BlobgalongDisplay.tsx [hunk 3]
```diff
@@ -75,11 +75,25 @@ export const BlobalongDisplay = () => {
 
       <div className="flex">
         {showMakeConnection && (
-          <button onClick={handleMakeConnection}>Make connection</button>
+          <button
+            onClick={handleMakeConnection}
+            // Prevent bubbling to Select machine event listeners
+            onMouseDown={(e) => e.stopPropagation()}
+            onMouseUp={(e) => e.stopPropagation()}
```
Comment (Thimey) [inline src/Game/SelectionDisplay/BlobgalongDisplay.tsx:82]: Can we just do this on the parent display component?

**Review by Thimey — APPROVED** [review]
Nice work! 💪

I think the next blob we add, we look at abstracting the config for common behaviour like selecting and movement
